barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_opcode.hpp

```diff
@@ -113,6 +117,18 @@ class Bytecode {
     static bool is_valid(uint8_t byte);
 };
 
+// Look into whether we can do something with concepts here to enable OpCode as a parameter
+template <typename T>
+    requires(std::unsigned_integral<T>)
+std::string to_hex(T value)
+{
+    std::ostringstream stream;
+    auto num_bytes = static_cast<uint64_t>(sizeof(T));
+    uint64_t mask = (static_cast<uint64_t>(1) << (num_bytes * 8)) - 1;
```

Comment:
Does the requires() enforce that T cannot be longer than 64 bits? 128-bit integers would not be supported.